File: identify_missing_timestamps.py
import os
import re

def is_description_file(filename):
    return filename.endswith('.description')

def has_timestamps(filepath):
    timestamp_pattern = re.compile(r'\d{2}:\d{2}')
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    return bool(timestamp_pattern.search(content))

def main(input_dir):
    for filename in os.listdir(input_dir):
        if is_description_file(filename):
            filepath = os.path.join(input_dir, filename)
            if not has_timestamps(filepath):
                print(f"No timestamps found in file: {filename}")

if __name__ == '__main__':
    base_dir = os.getcwd()
    input_dir = base_dir
    main(input_dir)

File: src/identify-missing-timestamps.ts
import * as fs from 'fs';
import * as path from 'path';

function isDescriptionFile(filename: string): boolean {
    return filename.endsWith('.description');
}

function hasTimestamps(filepath: string): boolean {
    const timestampPattern = /\d{2}:\d{2}/;
    const content = fs.readFileSync(filepath, 'utf-8');
    return timestampPattern.test(content);
}

function main(inputDir: string): void {
    const files = fs.readdirSync(inputDir);
    
    for (const filename of files) {
        if (isDescriptionFile(filename)) {
            const filepath = path.join(inputDir, filename);
            if (!hasTimestamps(filepath)) {
                console.log(`No timestamps found in file: ${filename}`);
            }
        }
    }
}

if (require.main === module) {
    const baseDir = process.cwd();
    main(baseDir);
}